staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -193,3 +194,127 @@ type JobCondition struct {
+	// +optional
+	Spec JobSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CronJob represents the configuration of a single cron job.
+type CronJob struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Specification of the desired behavior of a cron job, including the schedule.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Spec CronJobSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+	// Current status of a cron job.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Status CronJobStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CronJobList is a collection of cron jobs.
+type CronJobList struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// items is the list of CronJobs.
+	Items []CronJob `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// CronJobSpec describes how the job execution will look like and when it will actually run.
+type CronJobSpec struct {
+
+	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	Schedule string `json:"schedule" protobuf:"bytes,1,opt,name=schedule"`
+
+	// Optional deadline in seconds for starting the job if it misses scheduled
+	// time for any reason.  Missed jobs executions will be counted as failed ones.
+	// +optional
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty" protobuf:"varint,2,opt,name=startingDeadlineSeconds"`
+
+	// Specifies how to treat concurrent executions of a Job.
+	// Valid values are:
+	// - "Allow" (default): allows CronJobs to run concurrently;
+	// - "Forbid": forbids concurrent runs, skipping next run if previous run hasn't finished yet;
+	// - "Replace": cancels currently running job and replaces it with a new one
+	// +optional
+	ConcurrencyPolicy ConcurrencyPolicy `json:"concurrencyPolicy,omitempty" protobuf:"bytes,3,opt,name=concurrencyPolicy,casttype=ConcurrencyPolicy"`
+
+	// This flag tells the controller to suspend subsequent executions, it does
+	// not apply to already started executions.  Defaults to false.
+	// +optional
+	Suspend *bool `json:"suspend,omitempty" protobuf:"varint,4,opt,name=suspend"`
```

Comment:
Is there any technical reason to keep this as a pointer? Since the default is false, I don't think there is a need for this to be a boolean. Since we are graduating to v1, we can take the chance to simplify this and just make it a boolean. @liggitt wdyt?

Reply:
@apelisse what's the current state of server-side apply dealing of a field like this? If this was `bool`, defaulted to `false` and was `omitempty`, a false value (whether it was the default or explicit from the user) wouldn't get serialized.

Would that make server-side apply think there was a change if the user applied an explicit `false` and it differed from the serialized value (which would omit `false`)?

Reply:
ping @apelisse

Reply:
I'll have to write a test for that, that's an interesting use-case. I'm afraid it'll fall in the "implicit normalization kubernetes does to objects" that server-side apply doesn't really like.

Reply:
But that implicit normalization doesn't change if the field is a pointer or not. In both cases, during defaulting, we add a value. Unless you are referring to something else?

Reply:
If the boolean isn't a pointer, it'll constantly get dropped, which means that SSA will remove the ownership of that field (because it disappeared).

Reply:
I don't think it's terrible, it would prevent a conflict from happening if it's applied to false and later someone wants to apply true. The other way around will be fine.

Reply:
I need to test, it's possible that this will force an etcd rewrite on any apply ... :-/ I think I know how we could fix that.

Reply:
Any conclusions?

Reply:
Since there is doubt, I would just use the pointer if that's fine with performance (which I'm sure it is).

Reply:
FYI @adtac